controllers/secretproviderclasspodstatus_controller.go

```diff
@@ -259,11 +300,28 @@ func (r *SecretProviderClassPodStatusReconciler) patchSecretWithOwnerRef(ctx con
 	}
 
 	patch := client.MergeFromWithOptions(secret.DeepCopy(), client.MergeFromWithOptimisticLock{})
-	err = controllerutil.SetOwnerReference(spcPodStatus, secret, r.Scheme)
-	if err != nil {
-		return err
+	needsPatch := false
+
+	secretOwnerMap := make(map[string]types.UID)
+	for _, or := range secret.GetOwnerReferences() {
+		secretOwnerMap[or.Name] = or.UID
+	}
+
+	for i := range spcPodStatus {
+		if _, exists := secretOwnerMap[spcPodStatus[i].Name]; exists {
```

Comment:
This ensures all current spcps are added to the secret's ownerReferences if missing. What about when an old spcps ownerReference needs to be removed?

Reply:
The spcps contains the pod as owner reference. When the pod is deleted, the corresponding spcps is deleted as part of cascading delete and once spcps is deleted, garbage collector will remove the deleted spcps from the owner reference list

Reply:
That makes sense. Can we add a test for pod deletion to ensure the ownerReference is always cleaned up?

Reply:
This test helps us validate the cascade delete today - https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/test/bats/azure.bats#L172-L199

Reply:
I don't see a test that deletes the pod from a list of ownerReferences of a secret used by multiple owners.

Reply:
PR for testing secret with multiple owners - https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/309. The single owner is a subset of this new test.